python/dgl/graphbolt/impl/neighbor_sampler.py

```diff
@@ -134,42 +187,38 @@ def sample_subgraphs(self, seeds, seeds_timestamp):
             seeds = {
                 ntype: seeds.get(ntype, default_tensor) for ntype in ntypes
             }
-        for hop in range(num_layers):
-            subgraph = self.sampler(
-                seeds,
-                self.fanouts[hop],
-                self.replace,
-                self.prob_name,
+        minibatch._seed_nodes = seeds
+        minibatch.sampled_subgraphs = []
+        return minibatch
+
+    @staticmethod
+    def _set_input_nodes(minibatch):
+        minibatch.input_nodes = minibatch._seed_nodes
+        return minibatch
+
+    # pylint: disable=arguments-differ
+    def sampling_stages(
```

Comment:
Why not using `sampling_layers `to keep consistent?

Reply:
I think because we not only have sampling operations but also compaction operations. I thought stage describes these better than layers.

Reply:
Actually, you are right, those components are named SamplePerLayer and CompactPerLayer.

Reply:
But SamplePerLayer is not a layer, it is a stage. If it was a layer, I would have called it SampleLayer instead.

Reply:
![image](https://github.com/dmlc/dgl/assets/16190101/48552431-5607-4e36-8ecf-6cbdd755d8ce)

Reply:
If we find a way to improve the naming and things like that, we can do that in another PR. Merging this PR will help me make progress with #7039.